thor/reader.py

```diff
@@ -114,106 +114,41 @@ def getLastDate(self):
-                return errorMessage
-
-        stopLat = startLat
-        while self.netCDF.variables['rlat'][stopLat] < rotTo.item(1, 0):
-            if stopLat < self.latLen:
-                stopLat = stopLat + 1
-            else:
-                errorMessage["error"] += "to-latitude"
-                return errorMessage
-
-        startLong = 0
-        while self.netCDF.variables['rlon'][
-                startLong] < rotFrom.item(0, 0):
-            if startLong < self.lonLen:
-                startLong = startLong + 1
-            else:
-                errorMessage["error"] += "from-longitude"
-                return errorMessage
-
-        stopLong = startLong + 1
-        while self.netCDF.variables['rlon'][
-              stopLong] < rotTo.item(0, 0):
-            if stopLong < self.lonLen:
-                stopLong = stopLong + 1
-            else:
-                errorMessage["error"] += "to-longitude"
-                return errorMessage
-
-        return({"ok": True,
-                "data": [startLat,
-                         stopLat,
-                         startLong,
-                         stopLong]})
-
-    # -------------------------------------
-    def getAreaNew(self,
-                   fromLat,
-                   toLat,
-                   fromLong,
-                   toLong):
+    def overlap(self,
+                fromLat,
+                toLat,
+                fromLong,
+                toLong):
 
-        if fromLat < self.minLat:
+        # Checking overlap
+        if self.minLat > toLat:
             return {"ok": False,
-                    "error": "fromLat is smaller than " +
-                    "smallest file value, " +
-                    str(self.minLat)}
-        if toLat > self.maxLat:
+                    "error": "No overlap"}
+        if self.maxLat < fromLat:
             return {"ok": False,
-                    "error": "toLat is latger than " +
-                    "largest file value, " +
-                    str(self.maxLat)}
-        if fromLong < self.minLon:
+                    "error": "No overlap"}
+        if self.minLon > toLong:
             return {"ok": False,
-                    "error": "fromLong is smaller than " +
-                    "smallest file value, " +
-                    str(self.minLon)}
-        if toLong > self.maxLon:
-
+                    "error": "No overlap"}
+        if self.maxLon < fromLong:
             return {"ok": False,
-                    "error": "toLong is larger than " +
-                    "largest file value, " +
-                    str(self.maxLon)}
-
-        startLat = int(floor((fromLat - self.minLat) * self.latScale))
-        stopLat = int(floor((toLat - self.minLat) * self.latScale))
-        startLong = int(floor((fromLong - self.minLon) * self.lonScale))
-        stopLong = int(floor((toLong - self.minLon) * self.lonScale))
+                    "error": "No overlap"}
+
+        # Checking area
```

Comment:
Useless comment.  Describe that you set the new to and from values to the overlapping area between the requested area and this file.